cloudbuild/python-scripts/parse_logs.py

```diff
@@ -0,0 +1,510 @@
+    if is_metric_found:
+        return total_metric_count
+    raise RuntimeError(
+        f'[Log: parse_logs ERROR] Unable to find the "{metric_string}" in any of'
+        ' the logs'
+    )
+
+
+def run(
+    cluster_project_name,
+    cluster_name,
+    region,
+    job_id,
+    arg_project,
+    arg_dataset,
+    arg_table,
+    query,
+):
+    """Method that calls all the helper function to determine success of a job.
+
+    Args:
+      cluster_project_name: Project ID of the GCP project that contains the
+        cluster.
+      cluster_name: name of the cluster on which the dataproc job is running.
+      region: region in which the cluster runs.
+      job_id: JOB_Id of the dataproc job.
+      arg_project: Resource project id (from which rows are read)
+      arg_dataset: Resource dataset name (from which rows are read)
+      arg_table: Resource table name (from which rows are read)
+      query: String containing the query incase needs to be run by the connector.
+
+    Raises:
+      AssertionError: When the rows read by connector and in the BQ table do not
+        match.
+    """
+    # Get the temp bucket name.
+    cluster_temp_bucket = get_cluster_temp_bucket(
+        cluster_project_name, cluster_name, region
+    )
+    # Get the pattern of logs inside the temp bucket in GCS.
+    logs_pattern = get_logs_pattern(cluster_project_name, region, job_id)
+    # Read the blob and get the metric from them
+    metric = read_logs(cluster_temp_bucket, logs_pattern, query)
+    bq_table_rows = get_bq_table_row_count(
+        cluster_project_name, arg_project, arg_dataset, arg_table, query
+    )
+    if metric != bq_table_rows:
+        raise AssertionError('[Log: parse_logs ERROR] Rows do not match')
+
+
+def validate_arguments(
+    arguments_dictionary, required_arguments, acceptable_arguments
+):
+    for required_argument in required_arguments:
+        if required_argument not in arguments_dictionary:
+            raise UserWarning(
+                f'[Log: parse_logs ERROR] {required_argument} argument not provided'
+            )
+    for key, _ in arguments_dictionary.items():
+        if key not in acceptable_arguments:
+            raise UserWarning(
+                f'[Log: parse_logs ERROR] Invalid argument "{key}" provided'
+            )
+
+
+def main(argv: Sequence[str]) -> None:
+    acceptable_arguments = {
+        'job_id',
+        'project_id',
+        'cluster_name',
+        'region',
+        'project_name',
+        'dataset_name',
+        'table_name',
+        'query',
+    }
+    required_arguments = acceptable_arguments - {'query'}
+
+    if len(argv) > len(acceptable_arguments) + 1:
+        raise app.UsageError(
+            '[Log: parse_logs ERROR] Too many command-line arguments.'
+        )
+    elif len(argv) < len(required_arguments) + 1:
+        raise app.UsageError(
+            '[Log: parse_logs ERROR] Too less command-line arguments.'
+        )
```

Comment:
Given that you have a separate `validate arguments` method, this is not needed. Please remove.